node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -65,28 +80,73 @@ object Main {
 
     implicit val time: Time[Task] = effects.time
 
+    implicit val envVars: EnvVars[Task] = EnvVars.envVars
+
     val program = conf.command match {
       case Eval(files) => new ReplRuntime().evalProgram[Task](files)
       case Repl        => new ReplRuntime().replProgram[Task].as(())
-      case Deploy(phlo, phloPrice, validAfterBlock, privateKey, location) =>
-        DeployRuntime
-          .deployFileProgram[Task](
-            phlo,
-            phloPrice,
-            validAfterBlock,
-            privateKey,
-            location
-          )
+      case Deploy(
+          phlo,
+          phloPrice,
+          validAfterBlock,
+          maybePrivateKey,
+          maybePrivateKeyPath,
+          location
+          ) =>
+        (maybePrivateKey, maybePrivateKeyPath) match {
```

Comment:
This can be rewritten without code duplication and quite readable, I would say:

```scala
        def decryptPrivateKey(encryptedPrivateKeyPath : Path) =
          for {
            maybePassword <- EnvVars[Task].get(RNodeDeployerPasswordEnvVar)
            password <- maybePassword.map(p => p.pure[Task])
              .getOrElse(ConsoleIO[Task].readPassword("Password for private key file: "))

            privateKey <- Secp256k1.parsePemFile[Task](encryptedPrivateKeyPath, password)
          } yield privateKey

        val getPrivateKey =
          maybePrivateKey.map(_.pure[Task])
            .combineK(maybePrivateKeyPath.map(decryptPrivateKey))
            .sequence

        for {
          privateKey <- getPrivateKey
          _ <- DeployRuntime.deployFileProgram[Task](
            phlo,
            phloPrice,
            validAfterBlock,
            privateKey,
            location
          )
        } yield()
```